ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -859,6 +859,209 @@ CMitsubaLoader::SContext::tex_ass_type CMitsubaLoader::getTexture(SContext& ctx,
+			if (_node->plastic.distribution == CElementBSDF::RoughSpecularBase::ASHIKHMIN_SHIRLEY)
+				_1stdword |= static_cast<uint32_t>(_node->plastic.alphaV.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_ALPHA_V_TEX;
+			_1stdword |= static_cast<uint32_t>(_node->plastic.nonlinear) << bsdf::BITFIELDS_SHIFT_NONLINEAR;
+			break;
+		case bsdf::OP_COATING:
+			_1stdword |= _node->coating.distribution << bsdf::BITFIELDS_SHIFT_NDF;
+			_1stdword |= static_cast<uint32_t>(_node->coating.alpha.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_ALPHA_U_TEX;
+			if (_node->coating.distribution == CElementBSDF::RoughSpecularBase::ASHIKHMIN_SHIRLEY)
+				_1stdword |= static_cast<uint32_t>(_node->coating.alphaV.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_ALPHA_V_TEX;
+			_1stdword |= static_cast<uint32_t>(_node->coating.sigmaA.value.type == SPropertyElementData::INVALID) >> bsdf::BITFIELDS_SHIFT_SIGMA_A_TEX;
+			break;
+		case bsdf::OP_WARD:
+			_1stdword |= _node->ward.variant << bsdf::BITFIELDS_SHIFT_WARD_VARIANT;
+			_1stdword |= static_cast<uint32_t>(_node->ward.alphaU.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_ALPHA_U_TEX;
+			_1stdword |= static_cast<uint32_t>(_node->ward.alphaV.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_ALPHA_V_TEX;
+			break;
+		case bsdf::OP_BLEND:
+			switch (_node->type)
+			{
+			case CElementBSDF::Type::BLEND_BSDF:
+				_1stdword |= static_cast<uint32_t>(_node->blendbsdf.weight.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_WEIGHT_TEX;
+				break;
+			case CElementBSDF::Type::MASK:
+				_1stdword |= static_cast<uint32_t>(_node->mask.opacity.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_WEIGHT_TEX;
+				break;
+			case CElementBSDF::Type::MIXTURE_BSDF:
+				//always constant (not texture)
+				break;
+			default: break; //do not let warnings rise
+			}
+			break;
+		case bsdf::OP_DIFFTRANS:
+			_1stdword |= static_cast<uint32_t>(_node->difftrans.transmittance.value.type == SPropertyElementData::INVALID) << bsdf::BITFIELDS_SHIFT_SPEC_TRANS_TEX;
+			break;
+		default: break; //any other ones dont need any extra flags
+		}
+
+		bsdf::instr_t instr = _1stdword;
+		//TODO register encoding (2nd dword)
+		return instr;
+	};
+	push(bsdf, 0u);
+	while (!stack.empty())
+	{
+		auto& top = stack.top();
+		if ((top.instr_1st_dword & bsdf::INSTR_OPCODE_MASK) <= bsdf::OP_INVALID || top.visited)
+		{
+			ctx.instrBuffer.push_back(emitInstr(top.instr_1st_dword, top.bsdf));
+			//TODO gen struct and insert into BSDF buf
+			stack.pop();
+		}
+		else if (!top.visited)
+		{
+			top.visited = true;
+			switch (top.instr_1st_dword & bsdf::INSTR_OPCODE_MASK)
+			{
+			case bsdf::OP_BLEND:
+				push(top.bsdf->blendbsdf.bsdf[1], top.instr_1st_dword);
+			default:
+				push(top.bsdf->blendbsdf.bsdf[0], top.instr_1st_dword);
```

Comment:
ekhm bsdfs with no children will attempt to push their non-existent children, because of your weird `visited` design.

You can use two stacks, its not a sin (at least all the bsdf table entries will be there first before you start emitting instructions)
https://www.geeksforgeeks.org/iterative-postorder-traversal/